src/aspire/volume/symmetry_groups.py

```diff
@@ -46,17 +46,37 @@ def __str__(self):
         return f"{self.to_string}"
 
     @staticmethod
-    def from_string(symmetry, dtype):
+    def parse(symmetry, dtype):
         """
-        Takes a string, ie. 'C1', 'C7', 'D3', 'T', 'O', and returns a concrete
-        SymmetryGroup object.
+        Takes a SymmetryGroup instance or a string, ie. 'C1', 'C7', 'D3', 'T', 'O', and returns a concrete
+        SymmetryGroup object with the specified dtype.
 
-        :param symmetry: A string indicating the symmetry of a molecule.
+        :param symmetry: A string (or SymmetryGroup instance) indicating the symmetry of a molecule.
         :param dtype: dtype for rotation matrices.
         :return: Concrete SymmetryGroup object.
         """
 
+        if symmetry is None:
+            return IdentitySymmetryGroup(dtype=dtype)
+
+        if isinstance(symmetry, SymmetryGroup):
+            if symmetry.dtype != dtype:
+                logger.warning(f"Recasting SymmetryGroup with dtype {dtype}.")
```

Comment:
Wondering if this might be better as an actual warning.  When do you expect this situation might come up?

Reply:
I should say, that in this case this would do the reasonable thing and set the `symmetry_group` to have the same dtype as the source.

Reply:
Makes me wonder if the symmetry groups might have been fine just always being doubles and let the consumers cast down to floats.  Lot of trouble for those (sym) dtypes.  I might have led us astray there 😇 .

All right, sounds good, thanks!